modules/photo/include/opencv2/ccm.hpp

```diff
@@ -0,0 +1,520 @@
+    @param linear_type the method of linearization;\n
+                       default: @ref LINEARIZATION_GAMMA
+    */
+    CV_WRAP void setLinear(LINEAR_TYPE linear_type);
+
+    /** @brief set Gamma
+
+    @note only valid when linear is set to "gamma";\n
+
+    @param gamma the gamma value of gamma correction;\n
+                 default: 2.2;
+    */
+    CV_WRAP void setLinearGamma(const double& gamma);
+
+    /** @brief set degree
+        @note only valid when linear is set to
+       - @ref LINEARIZATION_COLORPOLYFIT
+       - @ref LINEARIZATION_GRAYPOLYFIT
+       - @ref LINEARIZATION_COLORLOGPOLYFIT
+       - @ref LINEARIZATION_GRAYLOGPOLYFIT
+
+    @param deg the degree of linearization polynomial;\n
+            default: 3
+
+    */
+    CV_WRAP void setLinearDegree(const int& deg);
+
+    /** @brief set SaturatedThreshold.
+                The colors in the closed interval [lower, upper] are reserved to participate
+                in the calculation of the loss function and initialization parameters
+    @param lower the lower threshold to determine saturation;\n
+            default: 0;
+    @param upper the upper threshold to determine saturation;\n
+                 default: 0
+    */
+    CV_WRAP void setSaturatedThreshold(const double& lower, const double& upper);
+
+    /** @brief set WeightsList
+    @param weights_list the list of weight of each color;\n
+                        default: empty array
+    */
+    CV_WRAP void setWeightsList(const Mat& weights_list);
+
+    /** @brief set WeightCoeff
+    @param weights_coeff the exponent number of L* component of the reference color in CIE Lab color space;\n
+                         default: 0
+    */
+    CV_WRAP void setWeightCoeff(const double& weights_coeff);
+
+    /** @brief set InitialMethod
+    @param initial_method_type the method of calculating CCM initial value;\n
+            default: INITIAL_METHOD_LEAST_SQUARE
+    */
+    CV_WRAP void setInitialMethod(INITIAL_METHOD_TYPE initial_method_type);
+
+    /** @brief set MaxCount
+    @param max_count used in MinProblemSolver-DownhillSolver;\n
+        Terminal criteria to the algorithm;\n
+                     default: 5000;
+    */
+    CV_WRAP void setMaxCount(const int& max_count);
+
+    /** @brief set Epsilon
+    @param epsilon used in MinProblemSolver-DownhillSolver;\n
+        Terminal criteria to the algorithm;\n
+                   default: 1e-4;
+    */
+    CV_WRAP void setEpsilon(const double& epsilon);
+
+    /** @brief make color correction */
+    CV_WRAP void run();
+
+    CV_WRAP Mat getCCM() const;
+    CV_WRAP double getLoss() const;
+    CV_WRAP Mat get_src_rgbl() const;
+    CV_WRAP Mat get_dst_rgbl() const;
+    CV_WRAP Mat getMask() const;
+    CV_WRAP Mat getWeights() const;
+
+    /** @brief Infer using fitting ccm.
+    @param img the input image.
+    @param islinear default false.
+    @return the output array.
+    */
+    CV_WRAP Mat infer(const Mat& img, bool islinear = false);
```

Comment:
There is some confusion with `run()` method. I propose to rename both with more verbose names.